### PR TITLE
added documentation to promisePoller.

### PR DESCRIPTION
**Pull request by DevinR528**
I hope this is helpful. sorry the the quotes changed i can put back to single if you want.  i ran tests and linting just to make sure i didn't fat finger anything.  i only wanted to add docs.

thanks Devin Ragotzy

### CODE REVIEW

**Comment by forivall** [comment]
The example is bad. You wouldn't call the `taskFn` when passing it to `promisePoller`, you would just pass the function.

File: src/lib/promise-poller.js [hunk 1]
```diff
@@ -1,35 +1,67 @@
...
+ * @param {Object} options that contains polling options
+ * @param {function} options.taskFn a function that must resolve or reject a promise or return false to stop polling.
+ * @param {Number} options.interval number of ms until taskFn runs again.
+ * @param {Number} [options.retries=5] number of times taskFn is attempted.
+ * @param {String} [options.strategy='fixed-interval'] one of 'fixed-interval', 'linear-backoff', 'exponential-backoff'.
+ * @param {Number} [options.masterTimeout=null] ms to reject taskFns promise regardless of retries.
+ * @returns {Promise} returns taskFn's resolved promise or array of rejections
+ * @example 
+ * promisePoller({
+ *   taskFn: yourFn(),
```
Comment (forivall) [inline src/lib/promise-poller.js:30]: It was just this line that was bad, the rest of the doc was good, IMO.

Comment (DevinR528) [reply src/lib/promise-poller.js]: Yup you are completely right that is bad should I redo it or is it not worth it.

**Comment by DevinR528** [comment]
Oh yeah you are completely right that is bad should I redo it or is it not worth it.

**Comment by forivall** [comment]
I'd say redo it, but leave out all the formatting changes. I used a small bit of your language in [the typescript definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38508)

**Pull request closed without merging**